distributed/worker.py

```diff
@@ -1207,18 +1206,20 @@ async def _register_with_scheduler(self) -> None:
             msg = response["message"] if "message" in response else repr(response)
             logger.error(f"Unable to connect to scheduler: {msg}")
             raise ValueError(f"Unexpected response from register: {response!r}")
-        else:
-            await asyncio.gather(
-                *(
-                    self.plugin_add(name=name, plugin=plugin)
-                    for name, plugin in response["worker-plugins"].items()
-                )
+
+        self.batched_stream.start(comm)
+        self.status = Status.running
```

Comment:
while this `self.status = Status.running` is redundant because `Server.start` sets it we need the status to be `running` before starting plugins and we need the `self.batched_stream` started before updating the status so `_send_worker_status_change`'s message isn't silently dropped

Reply:
yuck...but good catch.